static/src/javascripts-legacy/projects/common/modules/atoms/story-questions.js

```diff
@@ -177,13 +177,18 @@ define([
             }
         }
 
-        sendNewStyleInteraction(atomId.trim(), 'SUBSCRIBE', prefAnswerDelivery.value);
+        sendNewStyleInteraction(atomId.trim(), 'SUBSCRIBE', prefAnswerDelivery);
     }
 
     return {
         init: function() {
 
             var atomId = $('.js-storyquestion-atom-id').attr('id');
+            var answerDeliveryPrefContainer = document.getElementById('js-delivery-selection-body-' + atomId);
+
+            for (var i = answerDeliveryPrefContainer.children.length; i >= 0; i--) {
+                answerDeliveryPrefContainer.appendChild(answerDeliveryPrefContainer.children[Math.random() * i | 0]);
```

Comment:
Do we need to floor the random number? i.e.
```js
answerDeliveryPrefContainer.children[Math.floor(Math.random() * i) | 0]
```

Reply:
If you mentally unroll the loop (I do that during breakfast), you can imagine a scenario like this:
```
answerDeliveryPrefContainer.appendChild(answerDeliveryPrefContainer.children[0]);
answerDeliveryPrefContainer.appendChild(answerDeliveryPrefContainer.children[0]);
answerDeliveryPrefContainer.appendChild(answerDeliveryPrefContainer.children[0]);
answerDeliveryPrefContainer.appendChild(answerDeliveryPrefContainer.children[0]);
...
```
You need another container. I would do that with numbers instead of DOM nodes though:
```
var pool = [0, 1, 2];
var flush = [];
while (pool.length > 0) {
  var rand = Math.random() * pool.length | 0;
  flush.push(pool[rand]);
  pool.splice(rand, 1);
}
```